File: BackEnd/Application/UseCases/Tickets/UpdateTicket/UpdateTicketCommandHandler.cs
using Domain.Aggregates;
using Infrastructure.Repository;
using Infrastructure.UnitOfWork;
using MediatR;
using Microsoft.EntityFrameworkCore;

namespace Application.UseCases.Tickets.UpdateTicket;

public class UpdateTicketCommandHandler(IRepository<Ticket, Guid> ticketRepository, IRepository<User, Guid> userRepository, IUnitOfWork unitOfWork) : IRequestHandler<UpdateTicketCommand, bool>
{
    public async Task<bool> Handle(UpdateTicketCommand request, CancellationToken cancellationToken)
    {
        var users = await userRepository.Where(x => request.UserIds.Contains(x.Id)).ToListAsync(cancellationToken);
        var ticket = await ticketRepository.GetByIdAsync(request.Id, cancellationToken);
        if (ticket == null) return false;
        ticket.Update(request.TicketStatusId, request.Title, request.Description, users);
        await ticketRepository.UpdateAsync(ticket);
        await unitOfWork.SaveChangesAsync(cancellationToken);
        return true;
    }
}

File: flutter/lib/application/use_cases/tickets/update_ticket/update_ticket_command_handler.dart
import 'package:enterprise_management/application/use_cases/tickets/update_ticket/update_ticket_command.dart';
import 'package:enterprise_management/infrastructure/repositories/ticket_repository.dart';
import 'package:enterprise_management/shared_kernel/cqrs/command_handler.dart';
import 'package:enterprise_management/shared_kernel/result/result.dart';

class UpdateTicketCommandHandler extends ICommandHandler<UpdateTicketCommand, bool> {
  const UpdateTicketCommandHandler({required this._ticketRepository});
  final ITicketRepository _ticketRepository;
  @override
  Future<Result<bool>> handle(UpdateTicketCommand command) async {
    final res = await _ticketRepository.updateTicket(id: command.id, title: command.title, description: command.description);
    return Result.success(res);
  }

}